keyvault/client/src/main/java/com/azure/keyvault/SecretClient.java

```diff
@@ -0,0 +1,525 @@
+     * The list Secrets operation is applicable to the entire vault. However, only the base secret identifier and its attributes are provided in the response.
+     * Individual secret versions are not listed in the response. This operation requires the secrets/list permission.
+     *
+     * @return A {@link Flux} containing {@link SecretAttributes} of all the secrets in the vault.
+     */
+    public Flux<SecretAttributes> listSecretsAsync() {
+        return getPagedSecrets(service.getSecrets(vaultEndPoint, DEFAULT_MAX_PAGE_RESULTS, API_VERSION, ACCEPT_LANGUAGE));
+    }
+
+    /**
+     * Lists {@link DeletedSecret secrets} of the key vault.
+     * The get deleted secrets operation returns the secrets that have been deleted for a vault enabled for soft-delete.
+     * This operation requires the secrets/list permission.
+     *
+     * @return A {@link Flux} containing all of the {@link DeletedSecret deleted secrets} in the vault.
+     */
+    public Flux<DeletedSecret> listDeletedSecretsAsync() {
+        return getPagedDeletedSecrets(service.getDeletedSecrets(vaultEndPoint, DEFAULT_MAX_PAGE_RESULTS, API_VERSION, ACCEPT_LANGUAGE));
+    }
+
+    /**
+     * List all versions of the specified secret. The individual secret version response in the list is represented by {@link SecretAttributes}.
+     * The full secret identifier and attributes are provided in the response. No values are returned for the secrets.
+     * This operation requires the secrets/list permission.
+     *
+     * @param name The name of the secret.
+     * @throws NullPointerException thrown if name parameter is null
+     * @return A {@link Flux} containing {@link SecretAttributes} of all the versions of the specified secret in the vault. Flux is empty if secret with {@code name} does not exist in key vault
+     */
+    public Flux<SecretAttributes> listSecretVersionsAsync(String name) {
+        Objects.requireNonNull(name, "The Secret name cannot be null.");
+        return getPagedSecrets(service.getSecretVersions(vaultEndPoint, name, DEFAULT_MAX_PAGE_RESULTS, API_VERSION, ACCEPT_LANGUAGE));
+    }
+
+    /**
+     * List secrets in a specified key vault.
+     * The list secrets operation is applicable to the entire vault.
+     * However, only the base secret identifier and its attributes are provided in the response. Individual secret versions are not listed in the response.
+     * This operation requires the secrets/list permission.
+     *
+     * @param maxPageResults Maximum number of results to return in a page. If not specified, the service will return up to 25 results.
+     * @throws NullPointerException thrown if maxPageResults parameter is null
+     * @return A {@link Flux} containing {@link SecretAttributes} of all the secrets in the key vault.
+     */
+    public Flux<SecretAttributes> listSecretsAsync(int maxPageResults) {
+        Objects.requireNonNull(maxPageResults, "The maximum page results parameter cannot be null.");
```

Comment:
Is it possible for a primitive type, int to be null? I thought it would default to 0. What if I submit `-10`?

Reply:
Yeah, you can't have a null int, it's not possible for primitives to be null. `Integer` on the other hand, being a non-primitive reference type, can be null. A reasonable check would be to ensure that this is a positive value, but my comment still stands that I'm dubious that these 'maxPageResults' overloads are necessary at all.

Reply:
Valid point.
Updated with positive int check.
These overloads might get removed once we sync with scott schaab.